src/interfaces/records/handlers/records-read.ts

```diff
@@ -53,13 +53,13 @@ export class RecordsReadHandler implements MethodHandler {
 
     const newestRecordsWrite = newestExistingMessage as RecordsWriteMessage;
     try {
-      await recordsRead.authorize(tenant, await RecordsWrite.parse(newestRecordsWrite), this.messageStore);
+      await this.storageController.authorizeRecordsRead(tenant, recordsRead, await RecordsWrite.parse(newestRecordsWrite));
```

Comment:
so i thought about this a bit more, and i think it might actually be nicer to pass the `StorageController` through `RecordsRead.prototype.authorize` instead of `MessageStore`, as that way we can basically isolate `MessageStore` vs` DataStore` vs etc. inside `StorageController` by having everyone else use only the `StorageController` API

i think ultimately it's only used for `MessageStore.prototype.query` inside a few of the `ProtocolAuthorization` methods, so we should be able to replace `messageStore.query` with `storageController.queryMessages` and be good

apologies for sending you down a different path.  lmk if you think what we have now is better and we can discuss further :)

Reply:
+1. You beat me to it. I made a similar suggestions above.

Reply:
Addressed in https://github.com/TBD54566975/dwn-sdk-js/pull/338/commits/931ab6bd4e5079229f6b9663de5a442eedfe4983